### PR TITLE
Interface proposal

### PR DESCRIPTION
**Pull request by chrismytton**
This is the proposed interface for this library.

Possible point of discussion:

- The interface for the constructor - I think using keyword arguments makes sense, because it gives us the flexibility to add other constructor arguments. On the other hand having a positional argument for the `table` might make things a bit more concise for the simple case.
- The name of the method that returns the processed HTML string. I've gone with `#html_string`, which seems like a reasonable description of what's returned from that method. I thought about including "table" somewhere in the method name, but the whole class is about tables, so I don't think that's necessary.

### CODE REVIEW

**Review by tmtmtmtm — COMMENTED** [review]
I suspect we might want to have (at least) two different methods here: one to return raw HTML, and one to return a Nokogiri object to represent the new table. We're going to be using Nokogiri internally anyway, and in many uses we'll probably prefer getting that back, rather than having to reparse the HTML again.

(Plus, if we don't there's no real value in making this a class at all rather than just a simple function…)

On positional vs keyword arguments I don't think I really mind either way, though if we're keeping keyword arguments then perhaps something like `raw` would be better than `table`, as it being a table is implied anyway, and as you say we're already in table-overload territory?

**Comment by chrismytton** [comment]
I've pushed a few more commits to address the feedback. Note that the tests are failing because there isn't any actual code on this branch, but Travis is trying to run anyway.

File: README.md [hunk 1]
```diff
@@ -22,7 +63,35 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+To use the library create a new instance of `TableUnspanner::UnspannedTable`, passing it a `table` keyword argument that contains an HTML string representing your table. Then call the `#html_string` method which will return a string containing the processed HTML table.
```
Comment (tmtmtmtm) [inline README.md]: This still implies the only argument is `html_string`, when the example later implies that `nokogiri_node` is the primary method, with a fallback of `html_string`.

**Review by tmtmtmtm — APPROVED** [review]
:+1:

**Pull request merged**